.github/workflows/ci.yml

```diff
@@ -154,39 +154,50 @@ jobs:
-      - runner-group=postsubmit
-      - os-family=macOS
+    runs-on: macos-14
     env:
       BUILD_DIR: build-macos
     steps:
       - name: "Checking out repository"
         uses: actions/checkout@v4.1.7
+      - id: "gcp-auth"
+        name: "Authenticating to Google Cloud"
+        if: needs.setup.outputs.write-caches == 1
+        uses: "google-github-actions/auth@v1"
+        with:
+          token_format: "access_token"
+          credentials_json: "${{ secrets.IREE_OSS_GITHUB_RUNNER_BASIC_TRUST_SERVICE_ACCOUNT_KEY }}"
+          create_credentials_file: false
       - name: "Updating git submodules"
         run: git submodule update --init --jobs 8 --depth 1
-      - name: "Installing Python packages"
+      # There are multiple versions of Xcode and SDKs installed on the macOS runner.
+      # Select the latest Xcode app to enable using Metal offline toolchain.
+      - name: "Update Xcode command line tools path"
         run: |
-          python3 -m venv .venv
-          source .venv/bin/activate
-          python3 -m pip install -r runtime/bindings/python/iree/runtime/build_requirements.txt
+          sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
+          xcrun metal --version
+          xcrun metallib --version
+      - name: "Setting up Python"
+        uses: actions/setup-python@v5.1.0
+        with:
+          python-version: "3.10"
+          cache: "pip"
+      - name: "Installing Python packages"
+        run: pip install -r runtime/bindings/python/iree/runtime/build_requirements.txt
+      - name: "Installing requirements"
+        # We need coreutils for `realpath` used in scripts.
+        # We need bash because the default one on macOS is fairly old and lacks
+        # features we use in scripts.
+        run: brew install ninja ccache coreutils bash
+      # Finally: build and run tests.
       - name: "Building IREE"
         env:
-          IREE_READ_REMOTE_CCACHE: 0
-          IREE_WRITE_REMOTE_CCACHE: 0
-          IREE_READ_LOCAL_CCACHE: 1
-          IREE_WRITE_LOCAL_CCACHE: ${{ needs.setup.outputs.write-caches }}
-          # We'll remove the GITHUB_WORKSPACE directory after the job.
-          # Persist the cache by storing in the parent directory.
-          CCACHE_DIR: ${{ github.workspace }}/../.ccache
-          CCACHE_MAXSIZE: 30G
-        run: |
-          source .venv/bin/activate
-          bash ./build_tools/cmake/build_all.sh "${BUILD_DIR}"
+          IREE_WRITE_REMOTE_CCACHE: ${{ needs.setup.outputs.write-caches }}
+          IREE_CCACHE_GCP_TOKEN: ${{ steps.gcp-auth.outputs.access_token }}
+          CCACHE_NAMESPACE: github-macos-13
+        run: bash ./build_tools/cmake/build_all.sh "${BUILD_DIR}"
       - name: "Testing IREE"
-        run: |
-          source .venv/bin/activate
-          IREE_METAL_DISABLE=0 bash ./build_tools/cmake/ctest_all.sh "${BUILD_DIR}"
+        run: IREE_METAL_DISABLE=0 bash ./build_tools/cmake/ctest_all.sh "${BUILD_DIR}"
```

Comment:
Well, the build finished in just under 2 hours, but Metal isn't supported:

https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#limitations-for-arm64-macos-runners

> Nested-virtualization and Metal Performance Shaders (MPS) are not supported due to the limitation of Apple's Virtualization Framework.

We're mainly using this build to test that the Metal HAL passes tests, so the self-hosted runners won't work here :/